OOP/OOP-3/TrainReservation.java

```diff
@@ -0,0 +1,322 @@
+	public void chart()//To show all the trains of either of goods or passangers
+	{
+		System.out.println("TrainId	From	To	S/W	Ticket	Duration");
+		for(int i=0;i<trainInfo.size();i++)
+		System.out.println(trainInfo.get(i));
+	}
+	//this method is used for show the chart of trains after booking
+	public void updatedChart(int numberOfSeats,String trainNumber) 
+	{
+		int i,j;
+		if(trainType.equalsIgnoreCase("goods"))
+		{
+			
+			String temp1[]=new String[6];
+			StringBuffer sb=new StringBuffer();
+			this.numberOfSeats=numberOfSeats;
+			this.trainNumber=trainNumber;
+			for(i=0;i<trainInfo.size();i++)//Loop for subtracting the number of booked seats
+			{
+				
+				temp1=trainInfo.get(i).split(",");
+				if(temp1[0].equalsIgnoreCase(trainNumber)) 
+				{
+					int hold=Integer.parseInt(temp1[3])-numberOfSeats;
+					temp1[3]=Integer.toString(hold);
+					break;
+				}
+			}
+			
+			try
+			{
+				FileWriter fw=new FileWriter("Goods.txt");
+				for(int k=0;k<trainInfo.size();k++)
+				{
+					String temp2=trainInfo.get(k);
+					if(k==i)
+					{
+						for(j=0;j<6;j++)
+						{
+							fw.write(temp1[j]+",");
+						}
+						fw.write("\n");
+					}
+					else
+						fw.write(temp2+"\n");
+				}
+				fw.close();
+				String line;
+				FileReader fr=new FileReader("Goods.txt");
+				BufferedReader bf=new BufferedReader(fr);
+				while((line=bf.readLine())!=null)
+				{
+					System.out.println(line);
+				}
+			}catch(Exception obException){}
+			
+		}
+		else if(trainType.equalsIgnoreCase("Passanger"))
+		{
+			String temp1[]=new String[6];
+			StringBuffer sb=new StringBuffer();
+			this.numberOfSeats=numberOfSeats;
+			this.trainNumber=trainNumber;
+			for(i=0;i<trainInfo.size();i++)
+			{
+				
+				temp1=trainInfo.get(i).split(",");
+				if(temp1[0].equalsIgnoreCase(trainNumber)) 
+				{
+					int hold=Integer.parseInt(temp1[3])-numberOfSeats;
+					temp1[3]=Integer.toString(hold);
+					break;
+				}
+			}
+			try	
+			{
+				FileWriter fw=new FileWriter("Passanger.txt");
+				for(int k=0;k<trainInfo.size();k++)
+				{
+					String temp2=trainInfo.get(k);
+					if(k==i)
+					{
+						for(j=0;j<6;j++)
+						fw.write(temp1[j]+",");
+						fw.write("\n");
+					}
+					else
+						fw.write(temp2+"\n");
+				}
+				fw.close();
+				String line;
+				FileReader fr=new FileReader("Passanger.txt");
+				BufferedReader bf=new BufferedReader(fr);
+				while((line=bf.readLine())!=null)
+					System.out.println(line);
+				} catch(Exception e) {}
+			}
+		
+	}
+}
+//this class is used for checking the availability of required seats 
+class CheckAvailability
+{
+	Scanner sc;
+	String from,to;
+	ArrayList<String> trainInfo=new ArrayList<String>();
+	Iterator<String> itTrainInfo=trainInfo.iterator();
+	public CheckAvailability(ArrayList<String> trainInfo,String from,String to) 
+	{
+		this.trainInfo=trainInfo;	
+		this.from=from;
+		this.to=to;
+	}
+	public void checkSeatWeight(String from,String to)//to check required trains are available or not 
```

Comment:
ACTION:No point of passing these method parameter as  these are class level variable and you are initializing them in parameterized constructor.